packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -328,6 +328,65 @@ export interface CustomEndpointOptions {
   readonly hostedZone?: route53.IHostedZone;
 }
 
+export interface WindowStartTime {
+  /**
+   * The start hour of the window in Coordinated Universal Time (UTC), using 24-hour time.
+   * For example, 17 refers to 5:00 P.M. UTC.
+   *
+   * @default - 0
+   */
+  readonly hours: number;
+  /**
+   * The start minute of the window, in UTC.
+   *
+   * @default - 0
+   */
+  readonly minutes: number;
+}
+
+export interface OffPeakWindow {
+  /**
+   * A custom start time for the off-peak window, in Coordinated Universal Time (UTC).
+   * The window length will always be 10 hours, so you can't specify an end time.
+   * For example, if you specify 11:00 P.M. UTC as a start time, the end time will automatically be set to 9:00 A.M.
+   *
+   * @default - 00:00 UTC
+   */
+  readonly windowStartTime?: WindowStartTime;
+}
+
+/**
+ * Off-peak window settings for the domain.
+ * You can't disable the off-peak window for a domain after it's enabled.
+ */
+export interface OffPeakWindowOptions {
+  /**
+   * Specifies whether off-peak window settings are enabled for the domain.
+   *
+   * @default - true
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * Off-peak window settings for the domain.
+   *
+   * @default - default window start time is 00:00 UTC
+   */
+  readonly offPeakWindow?: OffPeakWindow;
```

Comment:
I don't think we need this to be an interface either. We should be able to get away with just `offPeakWindow?: OffPeakWindow`. If its specified, users probably want it to be enabled as well. In general I like to default to a single prop rather than specifying interfaces.